### PR TITLE
minor fixes and compact some code

### PR DESCRIPTION
**Pull request by latot**

### CODE REVIEW

File: src/RageSoundReader_FileReader.cpp [hunk 1]
```diff
@@ -50,7 +50,7 @@ RageSoundReader_FileReader *RageSoundReader_FileReader::TryOpenFile( RageFileBas
 		return nullptr;
 	}
 	OpenResult ret = Sample->Open( pFile );
-	pFile = nullptr; // Sample owns it now
+
```
Comment (kyzentun) [inline src/RageSoundReader_FileReader.cpp]: Setting pFile to nullptr ensures that anybody erroneously using it after this point runs into an obvious problem.  It's a common technique to make mistakes more visible.

Comment (latot) [inline src/RageSoundReader_FileReader.cpp]: Hi, deleted, thx about the explanation.

File: src/arch/MemoryCard/MemoryCardDriverThreaded_Linux.cpp [hunk 2]
```diff
@@ -308,7 +308,7 @@ void MemoryCardDriverThreaded_Linux::GetUSBStorageDevices( vector<UsbStorageDevi
 
 			char szScsiDevice[1024];
 			char szMountPoint[1024];
-			int iRet = sscanf( sLine.c_str(), "%s %s", szScsiDevice, szMountPoint );
+			int iRet = sscanf( sLine.c_str(), "%1023s %0123s", szScsiDevice, szMountPoint );
```
Comment (kyzentun) [inline src/arch/MemoryCard/MemoryCardDriverThreaded_Linux.cpp]: Typo of 1023 on the second format arg.

**Comment by kyzentun** [comment]
I fixed the typo and the conflict in NoteField myself.

**Comment by latot** [comment]
Okis, thx.

**Pull request merged**